### ISSUE TITLE
(refactor) Implement FilterService across all Registries

### ISSUE THREAD

**Issue by jniles** [third-culture-software/bhima#1628]
A new FilterService with a standard API was merged into `master` in https://github.com/IMA-WorldHealth/bhima-2.X/pull/1566.  We should roll all these filters out over every registry in BHIMA, so that we have a standardized development and user experience.

The registries that need to be tackled are:
 - [x]  Cash Payments Registry (https://github.com/IMA-WorldHealth/bhima-2.X/issues/1664)
 - [x] Patient Registry ( #1771)
 - [x] Invoice Registry (#1760)
 - [x] Voucher Registry (#1777)
 - [x] General Ledger ()
 - [x] Inventory Registry (#1789)
 - [x] Purchase Order Registry (#1907)
 - [x] Inventories in Stock ()
 - [x] Lots in Stock (#1938)
 - [x] Stock Movements (#1987)

Once these are complete, the maintenance burden on any one registry should be a lot lighter.

**Comment by jniles** [comment]
I'm going to attempt to update the Cash Payments Registry to see how easy this is.

**Comment by jniles** [comment]
This can be closed once #2020 and #2021 land.

**Comment by jniles** [comment]
They have landed.
